### PR TITLE
Loadtests for the M1 plugin

### PR DESCRIPTION
**Pull request by kajan-bolt**
# Description
These tests will load test the M1 plugin. 

Main tests that we are interested in: 
- Converting cart into Bolt Order
- Shipping and Tax webhook
- Preauth webhook 

We want to make sure that our plugin won't bring down a merchant's store during times of high load. 

Fixes: https://app.asana.com/0/941895179897714/1149771740841855/f

(Optional: Add a changelog by writing a comment after "#changelog" on the next line)

\#changelog

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [ x ] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [ ] Successfully tested locally (or docker image)
- [ x ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [ ] My code follows the style guidelines of this project.
- [ x ] I have performed a self-review of my own code.
- [ x ] I have commented my code, particularly in hard-to-understand areas.
- [ x ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

File: tests/load/config.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import crypto from "k6/crypto";
+
+export const URL = "http://107.22.129.12";
```
Comment (mccottry) [inline tests/load/config.js]: Consider adding a comment explaining this hard coded values  At minimum, can we specify that it is a place holder that needs to be replaced?

File: tests/load/config.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+export const CART_PATH = "/boltpay/cartDataLoadTest";
+export const SHIPPING_PATH = "/boltpay/shipping";
+export const PREAUTH_PATH = "/boltpay/api/create_order";
+const SIGNING_SECRET = "f527a8972d13c1ecdc31fd45a47054a249fe058b5bede0fc652dec986c95bfea";
```
Comment (mccottry) [inline tests/load/config.js]: Consider adding a comment explaining this hard coded values  At minimum, can we specify that it is a place holder that needs to be replaced?

File: tests/load/inputs/preauth_cart.js [hunk 3]
```diff
@@ -0,0 +1,24 @@
+import { create_amount } from "./amount.js";
+import { address } from "./address.js";
+import { create_test_item } from "./test_item.js";
+import { shipment } from "./shipment.js";
+
+export function create_preauth_cart ( order_reference ) {
+    return {
+        "order_reference": order_reference,
+        "display_id": "100044865|" + order_reference,
```
Comment (mccottry) [inline tests/load/inputs/preauth_cart.js]: In reality, the number after the pipe will not be the `order_reference` but rather a similar number that is greater than the `order_reference`, typically by 1 or 2.  I'm not sure this affects the test however.

File: tests/load/config.js (hunk 2 above)
Comment (kajan-bolt) [reply tests/load/config.js]: Sounds good, I added the comments

File: tests/load/inputs/preauth_cart.js (hunk 3 above)
Comment (kajan-bolt) [reply tests/load/inputs/preauth_cart.js]: Interesting, but yeah I double checked and it shouldn't affect the test

**Comment by matt-thomason** [comment]
> Preauth webhook

We should test non-preauth workflow as well, since many merchants have still not upgraded and it's our fallback path.

**Pull request merged**